### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by jnogle**
Added first data validation rules for certain collections, along with tests I've been able to run locally with the npm package called 'firestore-security-tests'...hopefully you can use these tests as a reference to further test the security rules you wrote earlier too @fedorareis

### CODE REVIEW

File: firestore.rules [hunk 1]
```diff
@@ -20,20 +20,76 @@ service cloud.firestore {
       return collection == "user";
     }
 
-    match /users/{userId} {
+    match /users/{id} {
```
Comment (fedorareis) [inline firestore.rules]: Is there a specific reason for changing it to id? If not then I prefer the more verbose name.

File: firestore.rules [hunk 2]
```diff
@@ -20,20 +20,76 @@ service cloud.firestore {
...
+
+    match /campuses/{id} {
+      // Data validation for required fields
+      allow write: if request.resource.data.keys().hasAll(['name', 'location'])
+                   && request.resource.data.name is string
+                   && request.resource.data.location is string
+                   && isAdmin();
+    }
+
+    match /communityGroups/{id} {
```
Comment (fedorareis) [inline firestore.rules]: See the previous comment about id.

File: firestore.rules [hunk 3]
```diff
@@ -20,20 +20,76 @@ service cloud.firestore {
...
       allow update: if isLeader();
```
Comment (fedorareis) [inline firestore.rules]: You are going to need explicit rules here also. the update rules should be mostly the same as the write rules with the condition that the leader info can't be changed.

**Review by fedorareis — CHANGES_REQUESTED** [review]
Respond to my comments on the rules file and you should be good to go. I'm going to trust that the test files work correctly and are good to go.

**Comment by jnogle** [comment]
Sounds good, I will make those changes and I am also going to make a second commit with the rest of the validation, so there will be a little more to look over

**Review by fedorareis — CHANGES_REQUESTED** [review]
The last change is to apply the id rule from the previous review to all of the models. So in general terms `match /models/{id}` should be `match /models/{modelsId}`

**Review by fedorareis — APPROVED** [review]

**Pull request merged**